query/dev/system/src/main/java/org/codice/ddf/admin/query/dev/system/fields/BundleField.java

```diff
@@ -0,0 +1,215 @@
+
+  public static final String IMPORTED_PKGS = "importedPkgs";
+
+  private IntegerField id;
+
+  private StringField name;
+
+  private StringField location;
+
+  private StringField state;
+
+  private PackageField.ListImpl exportedPkgs;
+
+  private PackageField.ListImpl importedPkgs;
+
+  private ServiceField.ListImpl services;
+
+  private ServiceReferenceField.ListImpl refs;
+
+  private ServiceReferenceListField.ListImpl refLists;
+
+  public BundleField() {
+    super(DEFAULT_FIELD_NAME, FIELD_TYPE_NAME, DESCRIPTION);
+    id = new IntegerField(BUNDLE_ID);
+    name = new StringField(BUNDLE_NAME);
+    location = new StringField(BUNDLE_LOCATION);
+    state = new StringField(BUNDLE_STATE);
+    services = new ServiceField.ListImpl();
+    exportedPkgs = new PackageField.ListImpl(EXPORTED_PKGS);
+    importedPkgs = new PackageField.ListImpl(IMPORTED_PKGS);
+    refs = new ServiceReferenceField.ListImpl();
+    refLists = new ServiceReferenceListField.ListImpl();
+  }
+
+  public BundleField id(int id) {
+    this.id.setValue(id);
+    return this;
+  }
+
+  public BundleField bundleName(String name) {
+    this.name.setValue(name);
+    return this;
+  }
+
+  public BundleField location(String location) {
+    this.location.setValue(location);
+    return this;
+  }
+
+  public BundleField addService(ServiceField ref) {
+    services.add(ref);
+    return this;
+  }
+
+  public BundleField addServiceRef(ServiceReferenceField ref) {
+    refs.add(ref);
+    return this;
+  }
+
+  public BundleField addServiceRefList(ServiceReferenceListField refList) {
+    refLists.add(refList);
+    return this;
+  }
+
+  public BundleField addExportedPackage(PackageField importedPkg) {
+    exportedPkgs.add(importedPkg);
+    return this;
+  }
+
+  public BundleField addImportedPackage(PackageField importedPkg) {
+    importedPkgs.add(importedPkg);
+    return this;
+  }
+
+  public Integer id() {
+    return id.getValue();
+  }
+
+  public String bundleName() {
+    return name.getValue();
+  }
+
+  public String location() {
+    return location.getValue();
+  }
+
+  public String state() {
+    return state.getValue();
+  }
+
+  public BundleField state(String state) {
+    this.state.setValue(state);
+    return this;
+  }
+
+  public BundleField state(int state) {
+    this.state.setValue(getBundleState(state));
+    return this;
+  }
+
+  public List<PackageField> exportedPackages() {
+    return exportedPkgs.getList();
+  }
+
+  public List<PackageField> importedPackages() {
+    return importedPkgs.getList();
+  }
+
+  public List<ServiceReferenceListField> serviceRefLists() {
+    return refLists.getList();
+  }
+
+  public List<ServiceReferenceField> serviceRefs() {
+    return refs.getList();
+  }
+
+  public List<ServiceField> services() {
+    return services.getList();
+  }
+
+  public static String getBundleState(int state) {
+    switch (state) {
+      case Bundle.UNINSTALLED:
```

Comment:
I definitely agree. I didn't want to spend the time writing them right now since the most value we get out of enum's is passing them as argument or programming against that enumerated set. Since we're doing neither, I'd rather until someone has one of those use classes.